reeprotocol/app_asdu.py

```diff
@@ -370,6 +371,75 @@ class M_IT_TG_2(M_IT_TX_2):
     type = 8
 
 
+class C_CB_UN_2(BaseAppAsdu):
+    """
+    Leer los bloques de totales integrados operacionales por intervalo de
+    tiempo para una dirección de objeto indicada
+    """
+    type = 190
+    data_length = 0x06
+    causa_tm = 6
+
+    def __init__(self, start_date=datetime.datetime.now(),
+                 end_date=datetime.datetime.now(), adr_object=10):
+        self.start_date = TimeA(start_date)
+        self.end_date = TimeA(end_date)
+        self.object = adr_object
+
+    def from_hex(self, data, cualificador_ev):
+        self.object = struct.unpack("B", data[0:1])[0]
+        self.start_date.from_hex(data[1:6])
+        self.end_date.from_hex(data[6:11])
+
+    def to_bytes(self):
+        response = bytearray()
+        response.extend(struct.pack("B", self.object))
+        response.extend(self.start_date.to_bytes())
+        response.extend(self.end_date.to_bytes())
+        return response
+
+    @property
+    def length(self):
+        return 0x14
+
+
+class M_IB_TK_2(BaseAppAsdu):
+    """
+    Bloques de totales integrados operacionales por intervalo de tiempo para una
+    dirección de objeto indicada
+    """
+    type = 140
+
+    def __init__(self):
+        self.valores = []
+        self.tiempo = None
+
+    def from_hex(self, data, cualificador_ev):
+        adr_aux = {
+            9: 8,
+            10: 6,
+            11: 3
+        }
+
+        position = 0
+        for i in range(0, cualificador_ev):
+            address = struct.unpack("B", data[position:position + 1])[0]
+            amount = adr_aux[address]
+            position += 1
```

Comment:
You can use `enumerare` to get the index inside the loop eg.

Also I don't see any usage of the `i` variable,  I will have the same value as position? Maybe you can do:

```python
for position in range(0, cualificador_ev):
...
```